packages/lib/src/basic-io.ts

```diff
@@ -0,0 +1,85 @@
+import { buildInputProcessor, buildOutputFormatter, IOFormat, itemTableFormatter, listTableFormatter, SmartThingsCommand,
+	TableFieldDefinition, writeOutput } from '.'
+
+
+// TODO: should these all be suffixed with "Function"? Probably
+export type GetData<O> = () => Promise<O>
+export type ListData<L> = () => Promise<L[]>
+export type LookupData<ID, O> = (id: ID) => Promise<O>
+export type ExecuteCommand<I, O> = (input: I) => Promise<O>
+
+
+// TODO: better names; e.g. TableOutputCommand => CommonOutputProducer or TableOutputProducer or similar
```

Comment:
I like the idea of using `Custom` for the `build...` version. I'm not 100% sold on keeping the word `Table` for the custom (build) one though since it can sometimes produce multiple tables and I could see it even producing something other than a table at some point. Going to stick with using table for now though unless we can think of something better.

Although, now that I think about it, maybe I like this better:

`TableCommonOutputProducer`
`CustomCommonOutputProducer`

(These are used for producing output in the `COMMON` format of `IOFormat`.)

Reply:
I ended up using `TableCommonOutputProducer` and `CustomCommonOutputProducer`